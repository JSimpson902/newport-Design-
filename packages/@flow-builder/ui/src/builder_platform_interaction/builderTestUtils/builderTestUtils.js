export * from './screenEditorTestUtils.js';
export * from './commonTestUtils';
export * from './deepFindMatchers';
export * from './useAdvancedOptionsTestUtils';
export * from './goldObjectMatchers';
export * from './domTestUtils';
export * from './complexTypeTestUtils';
