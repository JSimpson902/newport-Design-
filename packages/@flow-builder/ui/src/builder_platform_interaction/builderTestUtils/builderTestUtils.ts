export * from './combobox';
export * from './commonTestUtils';
export * from './complexTypeTestUtils';
export * from './deepFindMatchers';
export * from './domTestUtils';
export * from './events';
export * from './goldObjectMatchers';
export * from './manuallyAssignVariablesCheckboxTestUtils';
export * from './screenEditorTestUtils';
export * from './selectors';

// @sa11y testing appears to take some extra time (beyond the given 5000ms default jest timeout)
export const ACCESSIBILITY_TEST_TIMEOUT_IN_MS = 15000;

/**
 * Mocks a label in jest
 *
 * @param jest - The jest instance
 * @param labelName - The label name
 * @returns the mock label
 */
export function mockLabel(jest, labelName: string) {
    jest.mock(
        `@salesforce/label/${labelName}`,
        () => {
            return { default: `${labelName} mock value` };
        },
        { virtual: true }
    );

    return `${labelName} mock value`;
}
