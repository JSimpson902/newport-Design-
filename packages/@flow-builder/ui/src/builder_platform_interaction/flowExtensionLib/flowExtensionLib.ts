// @ts-nocheck
export * from './flowExtension';
