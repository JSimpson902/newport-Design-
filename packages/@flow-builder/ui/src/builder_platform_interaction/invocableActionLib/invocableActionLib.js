import {
    fetchOnce,
    SERVER_ACTION_TYPE
} from 'builder_platform_interaction/serverDataLib';
import {
    FLOW_AUTOMATIC_OUTPUT_HANDLING,
    getProcessTypeAutomaticOutPutHandlingSupport
} from 'builder_platform_interaction/processTypeLib';

let cachedDetails = [];
let invocableActions = [];

export function setInvocableActions(actions) {
    invocableActions = actions;
}

export function getInvocableActions() {
    return invocableActions;
}

export function fetchDetailsForInvocableAction(
    { actionName, actionType },
    { background = false, disableErrorModal = false, messageForErrorModal } = {}
) {
    const key = `${actionName}-${actionType}`;
    if (cachedDetails[key]) {
        return Promise.resolve(cachedDetails[key]);
    }
    const params = { actionName, actionType };
    return fetchOnce(
        SERVER_ACTION_TYPE.GET_INVOCABLE_ACTION_DETAILS,
        params,
        {
            background,
            disableErrorModal,
            messageForErrorModal
        }
    ).then(details => {
        cachedDetails[key] = details;
        return details;
    });
}

/**
 * Grabs the parameters for a specific invocable action from the cache
 *
 * @param {Object} actionId - the action identifier
 * @param {string} actionId.actionName - the action name
 * @param {string} actionId.actionType - the action type
 * @returns {Object} the action parameters
 */
export function getParametersForInvocableAction({ actionName, actionType }) {
    const key = `${actionName}-${actionType}`;
    return cachedDetails[key] && cachedDetails[key].parameters;
}

export function isAutomaticOutputHandlingSupported(flowProcessType) {
    const processTypeAutomaticOutPutHandlingSupport = getProcessTypeAutomaticOutPutHandlingSupport(
        flowProcessType
    );
    return (
        processTypeAutomaticOutPutHandlingSupport ===
        FLOW_AUTOMATIC_OUTPUT_HANDLING.SUPPORTED
    );
}

export function clearInvocableActionCachedParameters() {
    cachedDetails = [];
}
