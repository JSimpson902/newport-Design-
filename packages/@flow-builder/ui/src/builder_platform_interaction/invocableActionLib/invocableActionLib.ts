// @ts-nocheck
import { getValueFromHydratedItem } from 'builder_platform_interaction/dataMutationLib';
import { ParameterListRowItem } from 'builder_platform_interaction/elementFactory';
import {
    FLOW_AUTOMATIC_OUTPUT_HANDLING,
    getProcessTypeAutomaticOutPutHandlingSupport
} from 'builder_platform_interaction/processTypeLib';
import { fetchOnce, SERVER_ACTION_TYPE } from 'builder_platform_interaction/serverDataLib';

let cachedDetails = [];
let standardInvocableActions = [];
let dynamicInvocableActions = [];

export type InvocableAction = {
    elementType: ELEMENT_TYPE;
    actionType: string | ValueWithError;
    actionName: string | ValueWithError;
    inputParameters: ParameterListRowItem[];
    outputParameters: ParameterListRowItem[];
    displayText?: string | ValueWithError;
    name?: string | ValueWithError;
    label?: string | ValueWithError;
};
/**
 * Set standard invocable actions.
 *
 * @param actions - The standard actions to set
 */
export function setStandardInvocableActions(actions) {
    standardInvocableActions = actions;
}

/**
 * Set dynamic invocable actions.
 *
 * @param actions - The dynamic actions to set
 */
export function setDynamicInvocableActions(actions) {
    dynamicInvocableActions = actions;
}

/**
 * @returns standard invocable actions
 */
export function getStandardInvocableActions() {
    return standardInvocableActions;
}

/**
 * @returns dynamic invocable actions
 */
export function getDynamicInvocableActions() {
    return dynamicInvocableActions;
}

/**
 * @returns invocable actions
 */
export function getInvocableActions(): InvocableAction[] {
    return standardInvocableActions.concat(dynamicInvocableActions);
}

/**
 * @returns invocable actions and corresponding custom icon resource
 */
export function getActionIconMap() {
    return standardInvocableActions
        .concat(dynamicInvocableActions)
        .filter((action) => action.iconName)
        .reduce((acc, action) => {
            acc[action.name] = action.iconName;
            return acc;
        }, {});
}

/**
 * @param root0
 * @param root0.actionName
 * @param root0.actionType
 * @param root1
 * @param root1.background
 * @param root1.disableErrorModal
 * @param root1.messageForErrorModal
 */
export function fetchDetailsForInvocableAction(
    { actionName, actionType },
    { background = false, disableErrorModal = false, messageForErrorModal = undefined } = {}
) {
    const key = getActionKey(actionName, actionType);

    if (cachedDetails[key]) {
        return Promise.resolve(cachedDetails[key]);
    }
    const params = { actionName, actionType };
    return fetchOnce(SERVER_ACTION_TYPE.GET_INVOCABLE_ACTION_DETAILS, params, {
        background,
        disableErrorModal,
        messageForErrorModal
    }).then((details) => {
        cachedDetails[key] = details;
        return details;
    });
}

/**
 * Get the key for an action
 *
 * @param actionName - The action name
 * @param actionType - The action type
 * @returns  the action key, or undefined if not a valid action
 */
export function getActionKey(actionName?: string, actionType?: string): string | undefined {
    if (actionName == null || actionType == null) {
        return undefined;
    }

    return `${actionName}-${actionType}`;
}

/**
 * Updates a data type and sets a subtype in generically-typed extension parameters
 *
 * @param {*} parameters - Screen field parameters
 * @param {*} dynamicTypeMappings - A mapping of generic types to concrete types
 */
export function applyDynamicTypeMappings(parameters, dynamicTypeMappings) {
    if (!dynamicTypeMappings || dynamicTypeMappings.length === 0) {
        return parameters;
    }

    return parameters
        .map((parameter) => ({
            parameter,
            dynamicTypeMapping: dynamicTypeMappings.find(
                (dynamicTypeMapping) => parameter.sobjectType === getValueFromHydratedItem(dynamicTypeMapping.typeName)
            )
        }))
        .map(({ parameter, dynamicTypeMapping }) => {
            if (dynamicTypeMapping && dynamicTypeMapping.typeValue) {
                parameter = Object.assign({}, parameter, {
                    sobjectType: getValueFromHydratedItem(dynamicTypeMapping.typeValue)
                });
            }
            return parameter;
        });
}

/**
 * Grabs the parameters for a specific invocable action from the cache
 *
 * @param {Object} actionId - the action identifier
 * @param {string} actionId.actionName - the action name
 * @param {string} actionId.actionType - the action type
 * @param actionId.dataTypeMappings
 * @returns {Object} the action parameters
 */
export function getParametersForInvocableAction({ actionName, actionType, dataTypeMappings }) {
    const key = getActionKey(actionName, actionType);
    const params = cachedDetails[key] && cachedDetails[key].parameters;
    return applyDynamicTypeMappings(params, dataTypeMappings);
}

/**
 * @param flowProcessType
 */
export function isAutomaticOutputHandlingSupported(flowProcessType) {
    const processTypeAutomaticOutPutHandlingSupport = getProcessTypeAutomaticOutPutHandlingSupport(flowProcessType);
    return processTypeAutomaticOutPutHandlingSupport === FLOW_AUTOMATIC_OUTPUT_HANDLING.SUPPORTED;
}

/**
 *
 */
export function clearInvocableActionCachedParameters() {
    cachedDetails = [];
}
