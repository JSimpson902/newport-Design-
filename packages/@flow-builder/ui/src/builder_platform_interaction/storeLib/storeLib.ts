// @ts-nocheck
export { combinedReducer } from './combinedReducer';
export { createSelector } from './createSelector';
export { Store } from './store';
export { generateGuid } from './guidGenerator';
export { deepCopy } from './deepCopy';
export { isPlainObject } from './isPlainObject';
