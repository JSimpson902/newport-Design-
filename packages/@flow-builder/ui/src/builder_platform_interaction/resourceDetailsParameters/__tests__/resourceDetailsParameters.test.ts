// @ts-nocheck
import { setDocumentBodyChildren } from 'builder_platform_interaction/builderTestUtils';
import { describeExtension } from 'builder_platform_interaction/flowExtensionLib';
import { fetchDetailsForInvocableAction } from 'builder_platform_interaction/invocableActionLib';
import ResourceDetailsParameters from 'builder_platform_interaction/resourceDetailsParameters';
import { Store } from 'builder_platform_interaction/storeLib';
import { fetchActiveOrLatestFlowOutputVariables } from 'builder_platform_interaction/subflowsLib';
import { createElement } from 'lwc';
import {
    mockActionLocalActionInAutomaticOutputsModeResourceDetails,
    mockActionSubmitForApprovalAutomaticOutputsModeResourceDetails,
    mockExtensionScreenfieldAutomaticOutputsModeResourceDetails,
    mockSubflowInAutomaticOutputModeResourceDetails
} from 'mock/resourceDetailsData';
import { flowWithAllElementsUIModel } from 'mock/storeData';
import { flowExtensionDetails as mockFlowExtensionDetails } from 'serverData/GetFlowExtensionDetails/flowExtensionDetails.json';
import { flowWithActiveAndLatest as mockFlowWithActiveAndLatest } from 'serverData/GetFlowInputOutputVariables/flowWithActiveAndLatest.json';
import { localActionSampleActionDetails } from 'serverData/GetInvocableActionDetails/localActionSampleActionDetails.json';
import { submitForApprovalActionDetails as mockSubmitForApprovalActionDetails } from 'serverData/GetInvocableActionDetails/submitForApprovalActionDetails.json';

const createComponentUnderTest = (resourceDetails) => {
    const el = createElement('builder_platform_interaction-resource-details-parameters', {
        is: ResourceDetailsParameters
    });
    el.resourceDetails = resourceDetails;
    setDocumentBodyChildren(el);
    return el;
};

jest.mock('builder_platform_interaction/flowExtensionLib', () => {
    const actual = jest.requireActual('builder_platform_interaction/flowExtensionLib');
    return {
        describeExtension: jest.fn(() =>
            Promise.resolve(
                actual.createExtensionDescription('flowruntime:email', mockFlowExtensionDetails['flowruntime:email'])
            )
        )
    };
});

jest.mock('builder_platform_interaction/invocableActionLib', () => ({
    fetchDetailsForInvocableAction: jest.fn(() => Promise.resolve(mockSubmitForApprovalActionDetails))
}));

jest.mock('builder_platform_interaction/subflowsLib', () => {
    const actual = jest.requireActual('builder_platform_interaction/subflowsLib');
    return {
        fetchActiveOrLatestFlowOutputVariables: jest.fn((flowName) => {
            if (flowName === 'flowWithActiveAndLatest') {
                return Promise.resolve(
                    actual.getActiveOrLatestInputOutputVariables(mockFlowWithActiveAndLatest).outputVariables
                );
            }
            return Promise.reject(`No flow with name ${flowName}`);
        })
    };
});

jest.mock('builder_platform_interaction/storeLib', () => require('builder_platform_interaction_mocks/storeLib'));

const SELECTORS = { SPINNER: '.spinner' };

const EXPECTED_MOCK_ORDERED_PARAMETERS_FOR_LC_EMAIL_IN_AUTO_MODE = [
    {
        apiName: 'disabled',
        label: 'Disabled',
        description: 'Prevents the user from modifying or copying the value.',
        typeIconName: 'utility:crossfilter'
    },
    {
        apiName: 'label',
        label: 'Label',
        description: 'The label that appears above the email field.',
        typeIconName: 'utility:text'
    },
    {
        apiName: 'placeholder',
        label: 'Placeholder Text',
        description:
            "Text that appears in the field when it's empty. Use placeholder text to give users a hint about what to enter in the field.",
        typeIconName: 'utility:text'
    },
    {
        apiName: 'readonly',
        label: 'Read Only',
        description: 'Prevents the user from modifying the value, but not from copying it.',
        typeIconName: 'utility:crossfilter'
    },
    {
        apiName: 'required',
        label: 'Required',
        description: 'Requires the user to enter a value.',
        typeIconName: 'utility:crossfilter'
    },
    {
        apiName: 'value',
        label: 'Value',
        description:
            "To provide a default value, set this attribute's value. To use the user-entered value elsewhere in your flow, store this attribute's output value in a variable.",
        typeIconName: 'utility:text'
    }
];

const EXPECTED_MOCK_ORDERED_PARAMETERS_FOR_ACTION_SUBMIT_FOR_APPROVAL_IN_AUTO_MODE = [
    {
        apiName: 'instanceId',
        label: 'Instance ID',
        description: 'The ID of the approval process instance.',
        typeIconName: 'utility:text'
    },
    {
        apiName: 'instanceStatus',
        label: 'Instance Status',
        description:
            'The status of the approval. The valid values are "Approved," "Rejected," "Removed," or "Pending."',
        typeIconName: 'utility:text'
    },
    {
        apiName: 'newWorkItemIds',
        label: 'New Work Item IDs',
        description: 'An array of the ID(s) of the work item(s) created for the next step in this approval process.',
        typeIconName: 'utility:text'
    },
    {
        apiName: 'actorIds',
        label: 'Next Approver IDs',
        description: 'An array of the ID(s) of the next approver(s).',
        typeIconName: 'utility:text'
    },
    {
        apiName: 'entityId',
        label: 'Record ID',
        description: 'The ID of the record submitted for approval.',
        typeIconName: 'utility:text'
    }
];

const EXPECTED_MOCK_ORDERED_PARAMETERS_FOR_ACTION_LOCAL_ACTION_IN_AUTO_MODE = [
    {
        apiName: 'greeting',
        label: 'greeting',
        description: null,
        typeIconName: 'utility:text'
    },
    {
        apiName: 'subject',
        label: 'subject',
        description: null,
        typeIconName: 'utility:text'
    }
];

const EXPECTED_MOCK_ORDERED_PARAMETERS_FOR_SUBFLOW_IN_AUTO_MODE = [
    { apiName: 'accountOutput', label: 'accountOutput', typeIconName: 'utility:record_alt' },
    { apiName: 'accountOutputCollection', label: 'accountOutputCollection', typeIconName: 'utility:record_alt' },
    { apiName: 'carOutput', label: 'carOutput', typeIconName: 'utility:apex' },
    {
        apiName: 'carOutputCollection',
        label: 'carOutputCollection',
        typeIconName: 'utility:apex'
    },
    { apiName: 'inputOutput1', label: 'inputOutput1', typeIconName: 'utility:text' },
    { apiName: 'inputOutput2', label: 'inputOutput2', typeIconName: 'utility:text' },
    { apiName: 'output1', label: 'output1', typeIconName: 'utility:text' },
    { apiName: 'output2', label: 'output2', typeIconName: 'utility:text' },
    { apiName: 'output3', label: 'output3', typeIconName: 'utility:text' }
];

describe('Resource Details parameters', () => {
    let resourceDetailsParametersComponent;
    beforeAll(() => {
        Store.setMockState(flowWithAllElementsUIModel);
    });
    afterAll(() => {
        Store.resetStore();
    });
    describe('Extension (ie: ligthning component) screenfield in automatic outputs mode', () => {
        describe('No fetch exception', () => {
            beforeEach(() => {
                resourceDetailsParametersComponent = createComponentUnderTest(
                    mockExtensionScreenfieldAutomaticOutputsModeResourceDetails
                );
            });
            describe('Parameters fetch server call OK and NO error', () => {
                test('check "Parameters" details (via API)', () => {
                    const parameters = resourceDetailsParametersComponent.parameters;
                    expect(parameters).toEqual(EXPECTED_MOCK_ORDERED_PARAMETERS_FOR_LC_EMAIL_IN_AUTO_MODE);
                });
                test('check UI: icon names, tooltip, labels...(snapshot) parameters displayed', () => {
                    expect(resourceDetailsParametersComponent).toMatchSnapshot();
                });
            });
            describe('Parameters fetch server call OK but error', () => {
                beforeAll(() => {
                    describeExtension.mockImplementation(() =>
                        Promise.reject(new Error('An error occured during extension parameters fetching'))
                    );
                });
                test('check "Parameters" details (via API)', () => {
                    const parameters = resourceDetailsParametersComponent.parameters;
                    expect(parameters).toHaveLength(0);
                });
                test('check UI: icon names, tooltip, labels (snapshot) no parameters displayed', () => {
                    expect(resourceDetailsParametersComponent).toMatchSnapshot();
                });
            });
        });
        describe('Fetch exception', () => {
            test('spinner should not be displayed)', () => {
                describeExtension.mockImplementation(() => {
                    throw new Error('Runtime exception this time');
                });
                resourceDetailsParametersComponent = createComponentUnderTest(
                    mockExtensionScreenfieldAutomaticOutputsModeResourceDetails
                );
                const spinner = resourceDetailsParametersComponent.shadowRoot.querySelector(SELECTORS.spinner);
                expect(spinner).toBeNull();
            });
        });
    });

    describe('core action', () => {
        describe('"submit for approval" in automatic outputs mode', () => {
            describe('No fetch exception', () => {
                beforeEach(() => {
                    resourceDetailsParametersComponent = createComponentUnderTest(
                        mockActionSubmitForApprovalAutomaticOutputsModeResourceDetails
                    );
                });
                describe('Parameters fetch server call OK and NO error', () => {
                    test('check "Parameters" details (via API)', () => {
                        const parameters = resourceDetailsParametersComponent.parameters;
                        expect(parameters).toEqual(
                            EXPECTED_MOCK_ORDERED_PARAMETERS_FOR_ACTION_SUBMIT_FOR_APPROVAL_IN_AUTO_MODE
                        );
                    });
                    test('check UI: icon names, tooltip, labels...(snapshot) parameters displayed', () => {
                        expect(resourceDetailsParametersComponent).toMatchSnapshot();
                    });
                });
                describe('Parameters fetch server call OK but error', () => {
                    beforeAll(() => {
                        fetchDetailsForInvocableAction.mockImplementation(() =>
                            Promise.reject(new Error('An error occured during extension parameters fetching'))
                        );
                    });
                    test('check "Parameters" details (via API)', () => {
                        const parameters = resourceDetailsParametersComponent.parameters;
                        expect(parameters).toHaveLength(0);
                    });
                    test('check UI: icon names, tooltip, labels (snapshot) no parameters displayed', () => {
                        expect(resourceDetailsParametersComponent).toMatchSnapshot();
                    });
                });
            });
        });

        describe('"Local action" in automatic outputs mode', () => {
            describe('No fetch exception', () => {
                beforeAll(() => {
                    fetchDetailsForInvocableAction.mockImplementation(() =>
                        Promise.resolve(localActionSampleActionDetails)
                    );
                });
                beforeEach(() => {
                    resourceDetailsParametersComponent = createComponentUnderTest(
                        mockActionLocalActionInAutomaticOutputsModeResourceDetails
                    );
                });
                describe('Parameters fetch server call OK and NO error', () => {
                    test('check "Parameters" details (via API)', () => {
                        const parameters = resourceDetailsParametersComponent.parameters;
                        expect(parameters).toEqual(
                            EXPECTED_MOCK_ORDERED_PARAMETERS_FOR_ACTION_LOCAL_ACTION_IN_AUTO_MODE
                        );
                    });
                    test('check UI: icon names, tooltip, labels...(snapshot) parameters displayed', () => {
                        expect(resourceDetailsParametersComponent).toMatchSnapshot();
                    });
                });
                describe('Parameters fetch server call OK but error', () => {
                    beforeAll(() => {
                        fetchDetailsForInvocableAction.mockImplementation(() =>
                            Promise.reject(new Error('An error occured during extension parameters fetching'))
                        );
                    });
                    test('check "Parameters" details (via API)', () => {
                        const parameters = resourceDetailsParametersComponent.parameters;
                        expect(parameters).toHaveLength(0);
                    });
                    test('check UI: icon names, tooltip, labels (snapshot) no parameters displayed', () => {
                        expect(resourceDetailsParametersComponent).toMatchSnapshot();
                    });
                });
            });
        });
    });

    describe('Subflow in automatic outputs mode', () => {
        describe('No fetch exception', () => {
            beforeEach(() => {
                resourceDetailsParametersComponent = createComponentUnderTest(
                    mockSubflowInAutomaticOutputModeResourceDetails
                );
            });
            describe('Parameters fetch server call OK and NO error', () => {
                test('check "Parameters" details (via API)', () => {
                    const parameters = resourceDetailsParametersComponent.parameters;
                    expect(parameters).toEqual(EXPECTED_MOCK_ORDERED_PARAMETERS_FOR_SUBFLOW_IN_AUTO_MODE);
                });
                test('check UI: icon names, tooltip, labels...(snapshot) parameters displayed', () => {
                    expect(resourceDetailsParametersComponent).toMatchSnapshot();
                });
            });
            describe('Parameters fetch server call OK but error', () => {
                beforeAll(() => {
                    fetchActiveOrLatestFlowOutputVariables.mockImplementation(() =>
                        Promise.reject(new Error('An error occured during subflow parameters fetching'))
                    );
                });
                test('check "Parameters" details (via API)', () => {
                    const parameters = resourceDetailsParametersComponent.parameters;
                    expect(parameters).toHaveLength(0);
                });
                test('check UI: icon names, tooltip, labels (snapshot) no parameters displayed', () => {
                    expect(resourceDetailsParametersComponent).toMatchSnapshot();
                });
            });
        });
        describe('Fetch exception', () => {
            test('spinner should not be displayed)', () => {
                describeExtension.mockImplementation(() => {
                    throw new Error('Runtime exception this time');
                });
                resourceDetailsParametersComponent = createComponentUnderTest(
                    mockSubflowInAutomaticOutputModeResourceDetails
                );
                const spinner = resourceDetailsParametersComponent.shadowRoot.querySelector(SELECTORS.spinner);
                expect(spinner).toBeNull();
            });
        });
    });
});
