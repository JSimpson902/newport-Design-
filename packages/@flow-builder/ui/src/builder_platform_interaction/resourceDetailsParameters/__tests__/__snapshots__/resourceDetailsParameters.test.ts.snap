// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource Details parameters Extension (ie: ligthning component) screenfield in automatic outputs mode No fetch exception Parameters fetch server call OK and NO error check UI: icon names, tooltip, labels...(snapshot) parameters displayed 1`] = `
<builder_platform_interaction-resource-details-parameters>
  #shadow-root(open)
    <div
      class="resource-detail-parameters-body"
    >
      <div
        class="slds-text-title_caps slds-text-color_weak slds-p-vertical_x-small"
      >
        FlowBuilderResourceDetailsParameters.header
      </div>
      <ul
        class="slds-has-block-links test-list-section-items"
      >
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="Prevents the user from modifying or copying the value."
            >
              Disabled
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="The label that appears above the email field."
            >
              Label
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="Text that appears in the field when it's empty. Use placeholder text to give users a hint about what to enter in the field."
            >
              Placeholder Text
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="Prevents the user from modifying the value, but not from copying it."
            >
              Read Only
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="Requires the user to enter a value."
            >
              Required
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="To provide a default value, set this attribute's value. To use the user-entered value elsewhere in your flow, store this attribute's output value in a variable."
            >
              Value
            </span>
          </div>
        </li>
      </ul>
    </div>
</builder_platform_interaction-resource-details-parameters>
`;

exports[`Resource Details parameters Extension (ie: ligthning component) screenfield in automatic outputs mode No fetch exception Parameters fetch server call OK but error check UI: icon names, tooltip, labels (snapshot) no parameters displayed 1`] = `
<builder_platform_interaction-resource-details-parameters>
  #shadow-root(open)
</builder_platform_interaction-resource-details-parameters>
`;

exports[`Resource Details parameters Subflow in automatic outputs mode No fetch exception Parameters fetch server call OK and NO error check UI: icon names, tooltip, labels...(snapshot) parameters displayed 1`] = `
<builder_platform_interaction-resource-details-parameters>
  #shadow-root(open)
    <div
      class="resource-detail-parameters-body"
    >
      <div
        class="slds-text-title_caps slds-text-color_weak slds-p-vertical_x-small"
      >
        FlowBuilderResourceDetailsParameters.header
      </div>
      <ul
        class="slds-has-block-links test-list-section-items"
      >
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              accountOutput
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              accountOutputCollection
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              carOutput
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              carOutputCollection
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              inputOutput1
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              inputOutput2
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              output1
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              output2
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              output3
            </span>
          </div>
        </li>
      </ul>
    </div>
</builder_platform_interaction-resource-details-parameters>
`;

exports[`Resource Details parameters Subflow in automatic outputs mode No fetch exception Parameters fetch server call OK but error check UI: icon names, tooltip, labels (snapshot) no parameters displayed 1`] = `
<builder_platform_interaction-resource-details-parameters>
  #shadow-root(open)
</builder_platform_interaction-resource-details-parameters>
`;

exports[`Resource Details parameters core action "Local action" in automatic outputs mode No fetch exception Parameters fetch server call OK and NO error check UI: icon names, tooltip, labels...(snapshot) parameters displayed 1`] = `
<builder_platform_interaction-resource-details-parameters>
  #shadow-root(open)
    <div
      class="resource-detail-parameters-body"
    >
      <div
        class="slds-text-title_caps slds-text-color_weak slds-p-vertical_x-small"
      >
        FlowBuilderResourceDetailsParameters.header
      </div>
      <ul
        class="slds-has-block-links test-list-section-items"
      >
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              greeting
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span>
              subject
            </span>
          </div>
        </li>
      </ul>
    </div>
</builder_platform_interaction-resource-details-parameters>
`;

exports[`Resource Details parameters core action "Local action" in automatic outputs mode No fetch exception Parameters fetch server call OK but error check UI: icon names, tooltip, labels (snapshot) no parameters displayed 1`] = `
<builder_platform_interaction-resource-details-parameters>
  #shadow-root(open)
</builder_platform_interaction-resource-details-parameters>
`;

exports[`Resource Details parameters core action "submit for approval" in automatic outputs mode No fetch exception Parameters fetch server call OK and NO error check UI: icon names, tooltip, labels...(snapshot) parameters displayed 1`] = `
<builder_platform_interaction-resource-details-parameters>
  #shadow-root(open)
    <div
      class="resource-detail-parameters-body"
    >
      <div
        class="slds-text-title_caps slds-text-color_weak slds-p-vertical_x-small"
      >
        FlowBuilderResourceDetailsParameters.header
      </div>
      <ul
        class="slds-has-block-links test-list-section-items"
      >
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="The ID of the approval process instance."
            >
              Instance ID
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="The status of the approval. The valid values are \\"Approved,\\" \\"Rejected,\\" \\"Removed,\\" or \\"Pending.\\""
            >
              Instance Status
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="An array of the ID(s) of the work item(s) created for the next step in this approval process."
            >
              New Work Item IDs
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="An array of the ID(s) of the next approver(s)."
            >
              Next Approver IDs
            </span>
          </div>
        </li>
        <li
          class="slds-item"
        >
          <div
            class="slds-grid"
          >
            <lightning-icon
              class="slds-m-right_xx-small slds-m-top_xx-small"
            >
              #shadow-root(open)
            </lightning-icon>
            <span
              title="The ID of the record submitted for approval."
            >
              Record ID
            </span>
          </div>
        </li>
      </ul>
    </div>
</builder_platform_interaction-resource-details-parameters>
`;

exports[`Resource Details parameters core action "submit for approval" in automatic outputs mode No fetch exception Parameters fetch server call OK but error check UI: icon names, tooltip, labels (snapshot) no parameters displayed 1`] = `
<builder_platform_interaction-resource-details-parameters>
  #shadow-root(open)
</builder_platform_interaction-resource-details-parameters>
`;
